### PR TITLE
expander component

### PR DESCRIPTION
**Pull request by pflopez**
Screen Shot 2024-08-29 at 17 35 05

### CODE REVIEW

File: src/interface/src/styleguide/expander-section/expander-section.component.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<details [open]="isOpen">
```
Comment (pflopez) [inline src/interface/src/styleguide/expander-section/expander-section.component.html:1]: @lastminutediorama the expander icon its not the default arrow thingy.
Not sure if we can use this with custom icons.

Comment (pflopez) [reply src/interface/src/styleguide/expander-section/expander-section.component.html]: I ended up switching for just doing this manually. Didn't like how much I needed to override from the native details + summary elements in order for this to work with custom icons.

Comment (lastminutediorama) [reply src/interface/src/styleguide/expander-section/expander-section.component.html]: got it.

File: src/interface/src/styleguide/expander-section/expander-section.component.scss [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  cursor: pointer;
+  display: flex;
+  align-items: center;
+
+  &::before {
+    display: block;
+    width: 20px;
+    height: 20px;
+    content: '';
+    background: url('data:image/svg+xml,<svg width="5" height="10" viewBox="0 0 5 10" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M0.333313 9.16683V0.833496L4.49998 5.00016L0.333313 9.16683Z" fill="black" /></svg>') no-repeat 50% 50%;
```
Comment (lastminutediorama) [inline src/interface/src/styleguide/expander-section/expander-section.component.scss:16]: 👍

File: src/interface/src/styleguide/expander-section/expander-section.component.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  styleUrl: './expander-section.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ExpanderSectionComponent<T extends OptionType> {
+  @Input() isOpen?: boolean;
+  @Input() title: string = '';
+  @Input() groupName = '';
+  @Input() options: T[] = [];
+  @Input() defaultOption?: T;
+  @Output() optionSelected = new EventEmitter<T>();
```
Comment (lastminutediorama) [inline src/interface/src/styleguide/expander-section/expander-section.component.ts:26]: can we add comments to these? e.g., I'm not sure I fully understand the intent of groupName

Comment (lastminutediorama) [reply src/interface/src/styleguide/expander-section/expander-section.component.ts:26]: Ah sorry, I was reviewing this like it was out of draft -- no worries, add them when you're ready :D

**Review by lastminutediorama — APPROVED** [review]
lgtm

**Pull request merged**